sed/calibrator/hextof.py

```diff
@@ -0,0 +1,123 @@
+"""sed.calibrator.hextof module. Code for handling hextof specific transformations and
+calibrations.
+"""
+from typing import Sequence
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+import pandas as pd
+import dask.dataframe
+
+
+def unravel_8s_detector_time_channel(
+    df: dask.dataframe.DataFrame,
+    time_sector_column: str = "dldTimeAndSector",
+    tof_step_column: str = "dldTimeSteps",
+    sector_id_column: str = "dldSectorID",
+    config: dict = None,
+) -> dask.dataframe.DataFrame:
+    """Converts the 8s time in steps to time in steps and sectorID.
+
+    The 8s detector encodes the dldSectorID in the 3 least significant bits of the
+    dldTimeSteps channel.
+
+    Args:
+        sector_delays (Sequece[float], optional): Sector delays for the 8s time.
+            Defaults to config["dataframe"]["sector_delays"].
+    """
+    df = df.dropna(subset=[time_sector_column])
+    if time_sector_column is None:
+        if config is None:
+            raise ValueError("Either time_sector_column or config must be given.")
+        time_sector_column = config["dataframe"]["time_sector_column"]
+        if time_sector_column not in df.columns:
+            raise ValueError(f"Column {time_sector_column} not in dataframe.")
+    if tof_step_column is None:
+        if config is None:
+            raise ValueError("Either tof_step_column or config must be given.")
+        tof_step_column = config["dataframe"]["tof_step_column"]
+    if sector_id_column is None:
+        if config is None:
+            raise ValueError("Either sector_id_column or config must be given.")
+        sector_id_column = config["dataframe"]["sector_id_column"]
+
+    df[sector_id_column] = (df[time_sector_column] % 8).astype(np.int8)
+    df[tof_step_column] = (df[time_sector_column] // 8).astype(np.int32)
+    return df
+
+
+def align_8s_sectors(
+        df: dask.dataframe.DataFrame,
+        sector_delays: Sequence[float] = None,
+        sector_id_column: str = "dldSectorID",
+        tof_step_column: str = "dldTimeSteps",
```

Comment:
We have been discussing to not put any defaults in the code. How I have solved this for the other calibrators is to make classes for them, initialize them from the config, and place the defaults in the default config.